### PR TITLE
Bump Mailer from 1.23 to 1.32.1

### PR DESCRIPTION
**Pull request by basil**
Bumps [Mailer](https://github.com/jenkinsci/mailer-plugin) from 1.23 to 1.32.1.

### CODE REVIEW

**Review by res0nance — APPROVED** [review]
Should unblock #5173

**Review by timja — APPROVED** [review]

**Comment by timja** [comment]
This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Pull request merged**
